watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -184,17 +186,32 @@ def add_costing(m):
         doc="Cost of disposing of saline brine/ NF permeate",
     )
 
-    m.fs.dye_recovery_revenue = Expression(
-        expr=(
-            m.fs.zo_costing.utilization_factor
-            * m.fs.zo_costing.dye_mass_cost
-            * pyunits.convert(
-                m.fs.dye_retentate.flow_mass_comp[0, "dye"],
-                to_units=pyunits.kg / m.fs.zo_costing.base_period,
-            )
-        ),
-        doc="Savings from dye-retentate recovered back to the plant",
-    )
+    if dye_revenue == True:
+        m.fs.dye_value = Expression(
+            expr=(
+                m.fs.zo_costing.utilization_factor
+                * m.fs.zo_costing.dye_recovery_cost
+                * pyunits.convert(
+                    m.fs.dye_retentate.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            ),
+            doc="Profit from recovering dye",
+        )
+    # Note: this is multiplied by -1 since the sign is handled implicitly
+    elif dye_revenue == False:
```

Comment:
I could lump everything into one expression, but the two expressions aren't exactly the same. They have different signs (as you already mentioned), but if `dye_revenue == True`, it uses the dye recovery value and if `dye revenue == False`, it uses the dye disposal value. And the absolute value of the dye recovery value is not equal to the absolute value of the dye disposal value. So I felt like having one expression would be a bit too clunky, but if you still think otherwise, I could change it.